diff -urN core-compiz-0.8/kde/window-decorator-kde4/window.cpp core-compiz-0.8-patched/kde/window-decorator-kde4/window.cpp
--- core-compiz-0.8/kde/window-decorator-kde4/window.cpp	2014-02-13 02:59:55.537260913 +1100
+++ core-compiz-0.8-patched/kde/window-decorator-kde4/window.cpp	2014-02-13 02:57:01.000000000 +1100
@@ -936,6 +936,11 @@
 {
 }
 
+KDecoration::QuickTileMode KWD::Window::quickTileMode() const
+{
+    return KDecoration::QuickTileNone;
+}
+
 #endif
 
 
diff -urN core-compiz-0.8/kde/window-decorator-kde4/window.h core-compiz-0.8-patched/kde/window-decorator-kde4/window.h
--- core-compiz-0.8/kde/window-decorator-kde4/window.h	2014-02-13 02:59:55.538260929 +1100
+++ core-compiz-0.8-patched/kde/window-decorator-kde4/window.h	2014-02-13 02:56:54.000000000 +1100
@@ -148,6 +148,7 @@
 	virtual int tabCount() const;
 	virtual long tabId(int idx) const;
 	virtual void untab(long id, const QRect& newGeom);
+	virtual QuickTileMode quickTileMode() const;
 #endif
 
 	virtual bool eventFilter (QObject* o, QEvent* e);
